src/Symfony/Component/Validator/Mapping/Loader/AbstractLoader.php

```diff
@@ -78,6 +78,10 @@ protected function newConstraint(string $name, mixed $options = null): Constrain
             $className = self::DEFAULT_NAMESPACE.$name;
         }
 
+        if ((new \ReflectionClass($className))->getConstructor()->getNumberOfRequiredParameters() > 0 && \PHP_VERSION_ID >= 80100 && \is_array($options)) {
```

Comment:
PHP 8.1 will be required by Symfony 6.1. The check can be removed.

Reply:
Thanks, done.